src/decorator/array/ArrayUnique.ts

```diff
@@ -2,14 +2,19 @@ import { ValidationOptions } from '../ValidationOptions';
 import { buildMessage, ValidateBy } from '../common/ValidateBy';
 
 export const ARRAY_UNIQUE = 'arrayUnique';
+export type ArrayUniqueIdentifier<T = any> = (o: T) => any;
 
 /**
  * Checks if all array's values are unique. Comparison for objects is reference-based.
  * If null or undefined is given then this function returns false.
  */
-export function arrayUnique(array: unknown): boolean {
+export function arrayUnique(array: unknown[], identifier?: ArrayUniqueIdentifier): boolean {
   if (!(array instanceof Array)) return false;
 
+  if (identifier) {
+    array = array.map(o => (o != null ? identifier(o) : o));
```

Comment:
Why do you check for `null` here before calling the `identifier` function?

Reply:
check `null` before calling or check `null` in every `identifier` function. I just pick the former